app/queries/get_change_offer_options.rb

```diff
@@ -21,7 +21,7 @@ def available_courses(provider:)
   def available_study_modes(course:)
     CourseOption
       .selectable
-      .where(course: offerable_courses.find(course.id))
+      .where(course: offerable_courses.find_by(id: course.id))
```

Comment:
Now that the `open_on_apply` restriction has been removed, is this change needed - as it will enable to easily identify issues with non retrievable courses allocated to offers.

Reply:
Good question. I think I'd like to keep it for now as this will avoid the offer tab 404s we might get. If this proves to add some strange behaviour we can reinstate/revisit